gnucash/report/report-system/test/test-report-utilities.scm

```diff
@@ -41,4 +55,385 @@
 							      ACCT-TYPE-ASSET
 							      q-start-date q-end-date)))
 	;; 10 is the right number (5 days, two splits per tx)
-	(or (equal? 10 (length splits)) (begin (format #t "Fail, ~d splits, expected 10~%" (length splits)) #f))))))
+	(test-equal "length splits = 10"
+          10
+          (length splits))))))
+
+(define (teardown)
+  (gnc-clear-current-session))
+
+(define (test-list-ref-safe)
+  (test-begin "list-ref-safe")
+  (let ((lst '(1 2)))
+    (test-equal "list-ref-safe normal"
+      1
+      (list-ref-safe lst 0))
+    (test-equal "list-ref-safe out of bounds"
+      #f
+      (list-ref-safe lst 3)))
+  (test-end "list-ref-safe"))
+
+(define (test-list-set-safe)
+  (test-begin "list-set-safe")
+  (let ((lst (list 1 2)))
+    (list-set-safe! lst 1 3)
+    (test-equal "list-set-safe normal"
+      '(1 3)
+      lst)
+    (list-set-safe! lst 5 1)
+    (test-equal "list-set-safe out-of-bounds"
+      '(1 3 #f #f #f 1)
+      lst))
+  (test-end "list-set-safe"))
+
+(define (test-gnc:monetary->string)
+  (test-begin "gnc:monetary->string")
+  (let* ((book (gnc-get-current-book))
```

Comment:
gnc-get-current-book calls gnc-get-current-session and that creates a session if one doesn't already exist; it does so in C where Guile's garbage collection can't see it, so if you don't call gnc-clear-current-session at the end of the test, it leaks the session memory. The fix is to use (test-group-with-cleanup) instead of (test-begin) and use either your teardown function or (gnc-clear-current-session) as the cleanup function.